packages/frontend/src/features/semanticViewer/components/Header/HeaderView.tsx

```diff
@@ -45,12 +46,22 @@ export const HeaderView: FC<Props> = ({
         { open: openAddToDashboardModal, close: closeAddToDashboardModal },
     ] = useDisclosure(false);
 
-    const canManageSemanticViewerChart = user.data?.ability?.can(
+    const savedChartSpaceUserAccess = useMemo(() => {
+        const access: SpaceShare[] = [];
+
+        if (chart.space.userAccess) {
+            access.push(chart.space.userAccess);
+        }
+
+        return access;
+    }, [chart]);
```

Comment:
no useMemo required here

```suggestion
    const savedChartSpaceUserAccess = chart.space.userAccess ? [chart.space.userAccess] : [];
```

this should just work fine